JourneysSquad/JourneysSquad/CSVHandler.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+
+/// An array of strings representing a single row in a CSV file.
+typealias DataRecord = [String]
+
+/// Provides functionality to handle CSV files.
+struct CSVHandler {
+    /// Initializes a CSVHandler with a specified file name.
+    /// If the file cannot be created or located, this initializer returns `nil`.
+    /// - Parameter fileName: The name of the CSV file to be handled.
+    /// Don't need provide file extention to the name of file, ".csv" will be appended automatically.
+    init?(fileName: String) {
+        let fileManager = FileManager.default
+        guard let documentsDirectory = try? fileManager.url(
+            for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true
+        ) else { return nil }
+        let fileURL = documentsDirectory.appending(path: fileName).appendingPathExtension(defaultExtention)
+
+        if !fileManager.fileExists(atPath: fileURL.path) {
+            let createdSuccessfully = fileManager.createFile(atPath: fileURL.path, contents: nil)
+
+            if !createdSuccessfully {
+                return nil
+            }
+        }
+        filePath = fileURL
```

Comment:
nit: can we try to improve readability:
```suggestion
        let fileManager = FileManager.default

        let documentsDirectory = try? fileManager.url(
            for: .documentDirectory,
            in: .userDomainMask,
            appropriateFor: nil,
            create: true
        )

        let fileURL = documentsDirectory?
            .appendingPathComponent(fileName, isDirectory: false)
            .appendingPathExtension(defaultExtention)

       guard let fileURL = fileURL,
              fileManager.fileExists(atPath: fileURL.path),
              fileManager.createFile(atPath: fileURL.path, contents: nil)
        else {
            return nil
        }

        filePath = fileURL
```
?

Reply:
Additionally, you can extract the `documentsDirectory` into a constant within struct and make the code is even shorter.

Reply:
>  guard let fileURL = fileURL,
>               fileManager.fileExists(atPath: fileURL.path),
>               fileManager.createFile(atPath: fileURL.path, contents: nil)
@dersim-davaod 
I don't understand how these conditions can be met?
If the file doesn't exist, we will execute `else` block. 
If the file exists, why we need `.createFile`

Reply:
Fixed.

Reply:
@dersim-davaod 
If the file exist, the condition `!fileManager.fileExists(atPath: fileURL.path)` will return `false` and we will get into `else` block.

Reply:
May be we need: 
`fileManager.fileExists(atPath: fileURL.path) || fileManager.createFile(atPath: fileURL.path, contents: nil)`

Reply:
```swift
guard let fileURL = fileURL,
fileManager.fileExists(atPath: fileURL.path) ||
fileManager.createFile(atPath: fileURL.path, contents: nil)
```

something like that?

Reply:
@kirpichonok thanks!